services/scabbard/libscabbard/src/service/mod.rs

```diff
@@ -393,6 +421,30 @@ impl Service for Scabbard {
                     ))
                     .map_err(|err| ServiceError::UnableToHandleMessage(Box::new(err)))
             }
+            ScabbardMessage_Type::NEW_BATCH => {
+                match self.version {
+                    ScabbardVersion::V1 => {
```

Comment:
Is there any point where they negotiate their versions, or would these be the point at which an admin would see that the services have a version mismatch?

Reply:
No, the version is determined by the service arguments provided in the circuit definition. It's up to the circuit creator to ensure they're the same, which is handled automatically by the scabbard circuit template.

Reply:
It would be good if we could verify that sort of thing at circuit creation time, but that is probably beyond the scope of this PR.